app/controllers/accounts_controller.rb

```diff
@@ -11,11 +11,13 @@ def edit; end
 
   def update
     attach_avatar_for(account)
-    if account.update(ps[:account])
-      redirect_to account_path
-    else
-      render :edit
+    ActiveRecord::Base.transaction do
```

Comment:
It's too hard to read, can we make it more readable and clearly distinguish the kid's flow? Specifically, it's not clear from this code that it's about the case when current user is the kid account. Note, current user absense it too vague condition and it can mean anything including "nobody is updating an account" that looks too werid.

Reply:
Maybe postpone these changes into a separate PR?